src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs

```diff
@@ -47,9 +47,45 @@ private double setParameters(IUnformattedData data, IReadOnlyList<ColumnInfo> co
          ExtractGeneralParameters(keys, data, metaDataCategories, ref totalRank);
          ExtractNonQualifiedHeadings(keys, missingKeys, columnInfos, data, ref totalRank);
          setSecondaryColumnUnit(columnInfos);
+         setDimensionsForMappings(columnInfos);
          return totalRank;
       }
 
+      private void setDimensionsForMappings(IReadOnlyList<ColumnInfo> columnInfos)
+      {
+         foreach (var parameter in _parameters.OfType<MappingDataFormatParameter>())
+         {
+            var mappingColumn = parameter.MappedColumn;
+
+            if (mappingColumn?.Unit == null || mappingColumn.Dimension != null)
+               continue;
+
+            var concreteColumnInfo = columnInfos.FirstOrDefault(x => x.Name == parameter.MappedColumn.Name);
```

Comment:
FirstOrDefault means thius cam be null. Or you do not check for null. Therefor either a check is missing or it should be First

Reply:
True, I changed it to First since null will mean the configuration is wrong and it should actually throw.